lib/ansible/plugins/doc_fragments/shell_common.py

```diff
@@ -1,61 +1,64 @@
-# Copyright (c) 2017 Ansible Project
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 
 class ModuleDocFragment(object):
-
     # common shelldocumentation fragment
-    DOCUMENTATION = """
+    DOCUMENTATION = r'''
 options:
   remote_tmp:
     description:
-      - Temporary directory to use on targets when executing tasks.
+    - Temporary directory to use on targets when executing tasks.
+    type: path
```

Comment:
This line is responsible for the test failure.

Reply:
Using type `path` is causing the `~/` to be evaluated too early. It ends up resolving to `/root/.ansible/tmp` (the user running ansible-playbook) instead of `/home/ansibletest1/.ansible/tmp` (the become user for the task).

Reply:
But this is just documentation, I don't see how this would change a running test ?

Reply:
This is used by shell plugins, it's only module docs where the documentation does nothing. Plugin documentation is used by the config engine to get and set options.

Reply:
Hmmm, so DOCUMENTATION is a misnomer there.
But why would this only trip on FreeBSD ?

Reply:
Not sure why it didn't trip FreeBSD but plugin DOCUMENTATION blocks as used by the config system to get/set options for the plugin. This ensures our docs are always aligned with the actual config options for plugins.